### PR TITLE
build: add SSR testing setup

### LINKED ISSUE #2285: feat: support SSR

### PR DESCRIPTION
**Pull request by kyubisation**
This PR creates the basic setup of SSR testing, by creating a .ssr.spec.ts file for each component, containing a simple test.

Closes #2285

### CODE REVIEW

File: src/elements-experimental/pearl-chain-vertical-item/pearl-chain-vertical-item.ssr.spec.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import { SbbPearlChainVerticalItemElement } from './pearl-chain-vertical-item.js';
+
+describe(`sbb-pearl-chain-vertical-item ${fixture.name}`, () => {
+  let root: SbbPearlChainVerticalItemElement;
+
+  beforeEach(async () => {
+    root = await fixture<SbbPearlChainVerticalItemElement>(html`
+      <sbb-pearl-chain-vertical-item></sbb-pearl-chain-vertical-item>
+    `);
```
Comment (jeripeierSBB) [inline src/elements-experimental/pearl-chain-vertical-item/pearl-chain-vertical-item.ssr.spec.ts:13]: modules?

File: src/elements/clock/clock.ssr.spec.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import { fixture } from '../core/testing/private.js';
+
+import { SbbClockElement } from './clock.js';
+
+describe(`sbb-clock ${fixture.name}`, () => {
+  let root: SbbClockElement;
+
+  beforeEach(async () => {
+    root = await fixture(html`<sbb-clock></sbb-clock>`);
```
Comment (jeripeierSBB) [inline src/elements/clock/clock.ssr.spec.ts:12]: modules

File: src/elements/dialog/dialog-actions/dialog-actions.ssr.spec.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+import { fixture } from '../../core/testing/private.js';
+
+import { SbbDialogActionsElement } from './dialog-actions.js';
+
+describe(`sbb-dialog-actions ${fixture.name}`, () => {
+  let root: SbbDialogActionsElement;
+
+  beforeEach(async () => {
+    root = await fixture(html`<sbb-dialog-actions></sbb-dialog-actions>`);
```
Comment (jeripeierSBB) [inline src/elements/dialog/dialog-actions/dialog-actions.ssr.spec.ts:12]: modules

File: src/elements/dialog/dialog-content/dialog-content.ssr.spec.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+import { fixture } from '../../core/testing/private.js';
+
+import { SbbDialogContentElement } from './dialog-content.js';
+
+describe(`sbb-dialog-content ${fixture.name}`, () => {
+  let root: SbbDialogContentElement;
+
+  beforeEach(async () => {
+    root = await fixture(html`<sbb-dialog-content>Content</sbb-dialog-content>`);
```
Comment (jeripeierSBB) [inline src/elements/dialog/dialog-content/dialog-content.ssr.spec.ts:12]: modules

File: src/elements/dialog/dialog-title/dialog-title.ssr.spec.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+import { fixture } from '../../core/testing/private.js';
+
+import { SbbDialogTitleElement } from './dialog-title.js';
+
+describe(`sbb-dialog-title ${fixture.name}`, () => {
+  let root: SbbDialogTitleElement;
+
+  beforeEach(async () => {
+    root = await fixture(html`<sbb-dialog-title back-button>Title</sbb-dialog-title>`);
```
Comment (jeripeierSBB) [inline src/elements/dialog/dialog-title/dialog-title.ssr.spec.ts:12]: modules

File: src/elements/image/image.ssr.spec.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import { fixture } from '../core/testing/private.js';
+
+import { SbbImageElement } from './image.js';
+
+describe(`sbb-image ${fixture.name}`, () => {
+  let root: SbbImageElement;
+
+  beforeEach(async () => {
+    const url = `${location.protocol}//${location.host}/src/elements/clock/assets/sbb_clock_face.svg`;
+    root = await fixture(html`<sbb-image image-src=${url}></sbb-image>`);
```
Comment (jeripeierSBB) [inline src/elements/image/image.ssr.spec.ts:13]: modules

File: src/elements/link/block-link-button/block-link-button.ssr.spec.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        size="m"
+        negative
+        name="name"
+        type="submit"
+        form="formid"
+      >
+        <sbb-icon name="chevron-small-right-small" slot="icon"></sbb-icon>
+        Travelcards &amp; tickets.
+      </sbb-block-link-button>`,
+    );
```
Comment (jeripeierSBB) [inline src/elements/link/block-link-button/block-link-button.ssr.spec.ts:24]: modules

File: src/elements/link/block-link-static/block-link-static.ssr.spec.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+      html` <sbb-block-link-static icon-placement="end" size="m">
+        <sbb-icon
+          aria-hidden="true"
+          data-namespace="default"
+          name="chevron-small-right-small"
+          role="img"
+          slot="icon"
+        ></sbb-icon>
+        Travelcards &amp; tickets.
+      </sbb-block-link-static>`,
```
Comment (jeripeierSBB) [inline src/elements/link/block-link-static/block-link-static.ssr.spec.ts:22]: modules

File: src/elements/link/block-link/block-link.ssr.spec.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  beforeEach(async () => {
+    root = await fixture(html`
+      <sbb-block-link
+        href="https://github.com/sbb-design-systems/lyne-components"
+        size="m"
+        download
+        accessibility-label="Travelcards &amp; tickets"
+      >
+        Travelcards &amp; tickets.
+      </sbb-block-link>
```
Comment (jeripeierSBB) [inline src/elements/link/block-link/block-link.ssr.spec.ts:20]: modules

File: src/elements/overlay/overlay.ssr.spec.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+import { SbbOverlayElement } from './overlay.js';
+
+describe(`sbb-overlay ${fixture.name}`, () => {
+  let root: SbbOverlayElement;
+
+  beforeEach(async () => {
+    root = await fixture(html`
+      <sbb-overlay id="my-overlay-1" accessibility-label="Label">
+        <p>Overlay content</p>
+      </sbb-overlay>
```
Comment (jeripeierSBB) [inline src/elements/overlay/overlay.ssr.spec.ts:15]: modules

File: src/elements/radio-button/radio-button-group/radio-button-group.ssr.spec.ts [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  beforeEach(async () => {
+    root = await fixture(html`
+      <sbb-radio-button-group value="Value one">
+        <sbb-radio-button id="sbb-radio-1" value="Value one">Value one</sbb-radio-button>
+        <sbb-radio-button id="sbb-radio-2" value="Value two">Value two</sbb-radio-button>
+        <sbb-radio-button id="sbb-radio-3" value="Value three" disabled
+          >Value three</sbb-radio-button
+        >
+        <sbb-radio-button id="sbb-radio-4" value="Value four">Value four</sbb-radio-button>
+      </sbb-radio-button-group>
```
Comment (jeripeierSBB) [inline src/elements/radio-button/radio-button-group/radio-button-group.ssr.spec.ts:22]: modules

File: src/elements/radio-button/radio-button/radio-button.ssr.spec.ts [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+
+import { fixture } from '../../core/testing/private.js';
+
+import { SbbRadioButtonElement } from './radio-button.js';
+
+describe(`sbb-radio-button ${fixture.name}`, () => {
+  let root: SbbRadioButtonElement;
+
+  beforeEach(async () => {
+    root = await fixture(html`<sbb-radio-button value="Value">Value label</sbb-radio-button>`);
```
Comment (jeripeierSBB) [inline src/elements/radio-button/radio-button/radio-button.ssr.spec.ts:12]: modules

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
LGTM, please add missing modules

**Review by TomMenga — APPROVED** [review]
LGTM 👍

**Pull request merged**